src/lib/Handshake.js

```diff
@@ -311,11 +320,7 @@ class Handshake {
 
     this._handshakeStage = 'send-session-key';
```

Comment:
yeah..also I think, we can get rid of this._handshakeFail and make it all with throwing errors instead of rejecting 3rd promise.